pkg/controller/apiserver/apiserver_controller.go

```diff
@@ -306,21 +305,23 @@ func (r *ReconcileAPIServer) Reconcile(ctx context.Context, request reconcile.Re
 			return reconcile.Result{}, err
 		}
 
-		if managementCluster != nil {
-			// TODO: Does this belong here or in the manager?
-			tunnelCASecret, err := utils.GetSecret(ctx, r.client, render.VoltronTunnelSecretName, common.OperatorNamespace())
+		if managementCluster != nil && !r.multiTenant {
+			// The secret that contains the CA x509 certificate to create client certificates for the managed cluster
+			// is created by the Manager controller in tigera-operator namespace. We will read this secret and make
+			// sure it is available in the same namespace as the API server (tigera-system)
+			// This secret is only created for a management cluster in a multi-cluster setup for a single tenant.
+			// Other cluster types do not require this secret. (Standalone configuration do not need it and multi-tenant
+			// configuration create secrets inside the tenant namespaces)
+			tunnelSecretName := managementCluster.Spec.TLS.SecretName
+			tunnelCASecret, err := utils.GetSecret(ctx, r.client, tunnelSecretName, common.OperatorNamespace())
 			if err != nil {
 				r.status.SetDegraded(operatorv1.ResourceReadError, "Unable to fetch the tunnel secret", err, reqLogger)
 				return reconcile.Result{}, err
 			}
 			if tunnelCASecret == nil {
-				tunnelCASecret, err = certificatemanagement.CreateSelfSignedSecret(render.VoltronTunnelSecretName, common.OperatorNamespace(), "tigera-voltron", []string{"voltron"})
-				if err != nil {
-					r.status.SetDegraded(operatorv1.ResourceCreateError, "Unable to create the tunnel secret", err, reqLogger)
-					return reconcile.Result{}, err
-				}
+				r.status.SetDegraded(operatorv1.ResourceReadError, "Unable to fetch the tunnel secret", err, reqLogger)
+				return reconcile.Result{}, err
 			}
-			tunnelSecretPassthrough = render.NewPassthrough(tunnelCASecret)
```

Comment:
Don't we need this passthrough in order to copy the secret into the right namespace?

Reply:
Where? I don't see that in this file.

Reply:
@caseydavenport I think it already is in master: https://github.com/tigera/operator/blob/master/pkg/controller/apiserver/apiserver_controller.go#L403